### PR TITLE
New tutorial: Volume coupled diffusion

### PR DESCRIPTION
**Pull request by BenjaminRodenberg**
This tutorial shows how to implement volume coupling using the FEniCS adapter.

### CODE REVIEW

File: volume-coupled-diffusion/fenics/volume-coupled-diffusion.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+
+u_n = interpolate(u_ini, V)
+
+
+dt = precice.initialize(AllDomain(), read_function_space=V, write_object=u_n)
+volume_term = precice.create_coupling_expression()
+f = Function(V)
+
+print(precice._fenics_vertices)
```
Comment (IshaanDesai) [inline volume-coupled-diffusion/fenics/volume-coupled-diffusion.py:55]:
```suggestion

```

**Review by IshaanDesai — APPROVED** [review]
The case runs and output is reasonable :+1: 
At the end of the simulation the following error is seen:
```
Error when destroying communicator (MPI_Comm_free).
```

File: volume-coupled-diffusion/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
+---
+title: Volume coupled diffusion
+permalink: tutorials-volume-coupled-diffusion.html
+keywords: FEniCS, Diffusion, Volume Coupling
+summary: This tutorial illustrates volume coupling via a simple example.
+---
+
+{% include note.html content="Get the [case files of this tutorial](https://github.com/benjaminrodenberg/tutorials/tree/volume-coupled-diffusion/volume-coupled-diffusion). Read how in the [tutorials introduction](https://www.precice.org/tutorials.html)." %}
```
Comment (uekerman) [inline volume-coupled-diffusion/README.md:8]:
```suggestion
{% include note.html content="Get the [case files of this tutorial](https://github.com/precice/tutorials/tree/master/volume-coupled-diffusion). Read how in the [tutorials introduction](https://www.precice.org/tutorials.html)." %}
```
If we want to merge it (actually, why not)

File: volume-coupled-diffusion/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+---
+title: Volume coupled diffusion
+permalink: tutorials-volume-coupled-diffusion.html
+keywords: FEniCS, Diffusion, Volume Coupling
+summary: This tutorial illustrates volume coupling via a simple example.
```
Comment (uekerman) [inline volume-coupled-diffusion/README.md:5]:
```suggestion
summary: This tutorial illustrates volume coupling with a simple example.
```

File: volume-coupled-diffusion/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+---
+
+{% include note.html content="Get the [case files of this tutorial](https://github.com/benjaminrodenberg/tutorials/tree/volume-coupled-diffusion/volume-coupled-diffusion). Read how in the [tutorials introduction](https://www.precice.org/tutorials.html)." %}
+
+## Setup
+
+This case illustrates how to implement volume coupling in a simple toy problem. Two diffusion problems are coupled via volume coupling. One domain (the source) has constant non-zero Dirichlet boundary conditions. The Other domain (the drain) has Neumann boundary conditions and a zero Dirichlet boundary condition at the right edge of the domain. The quantity u flows from the source to the drain. This is realized via volume coupling:
```
Comment (uekerman) [inline volume-coupled-diffusion/README.md:12]:
```suggestion
This case illustrates how to implement volume coupling in a simple toy problem. Two diffusion problems are coupled via volume terms. One domain (the source) has constant non-zero Dirichlet boundary conditions. The other domain (the drain) has Neumann boundary conditions and a zero Dirichlet boundary condition at the right edge of the domain. The quantity u flows from the source to the drain.
```

File: volume-coupled-diffusion/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+![Case setup of volume-coupled-diffusion case](images/tutorials-volume-coupled-diffusion-setup.png)
+
+## Available solvers and dependencies
+
+You can either couple a solver with itself or different solvers with each other. In any case you will need to have preCICE and the python bindings installed on your system.
```
Comment (uekerman) [inline volume-coupled-diffusion/README.md:18]:
```suggestion
```

File: volume-coupled-diffusion/README.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+* FEniCS. Install [FEniCS](https://fenicsproject.org/download/) and the [FEniCS-adapter](https://github.com/precice/fenics-adapter).
+
+## Running the simulation
+
+This tutorial is for FEniCS. You can find the corresponding `run.sh` script in the folder `fenics`.
+
+For choosing whether you want to create a source or drain, please provide the following commandline input:
```
Comment (uekerman) [inline volume-coupled-diffusion/README.md:26]:
```suggestion
To choose whether you want to run the source or the drain solver, please provide the following command line input:
```

**Review by uekerman — APPROVED** [review]

**Pull request merged**